src/plugin/fieldViews/ProseMirrorFieldView.ts

```diff
@@ -312,6 +313,29 @@ export abstract class ProseMirrorFieldView implements FieldView<string> {
     (this.innerEditorView.dom as HTMLDivElement).style.overflow = "auto";
   }
 
+  /**
+   * We need to set the appropriate selection in the parent editor when this field is focused.
+   * Prosemirror does not dispatch a transaction when it receives focus on an empty document,
+   * so in that case we do this manually.
+   */
+  private setupFocusHandler() {
+    this.fieldViewElement.addEventListener("focusin", () => {
+      if (!this.innerEditorView) {
+        return;
+      }
+      if (this.innerEditorView.state.doc.textContent.length !== 0) {
+        return;
+      }
+      const { tr, doc } = this.innerEditorView.state;
+      tr.setSelection(TextSelection.create(doc, 0));
+      // Setting a text selection seems to clear out our stored marks,
+      // so we must add them to the transaction explicitly.
+      tr.setStoredMarks(this.innerEditorView.state.storedMarks ?? []);
```

Comment:
Unfamiliar with `storedMarks`, what would be the consequence of them being removed?

Reply:
Stored marks [store styling information to be applied on the next input](https://prosemirror.net/docs/ref/#state.EditorState.storedMarks). It's the state that stores e.g. 'impending boldness' – for example, when you hit a keyboard shortcut or click a menu item to apply bold styling, the editor stores that state so bold styling is applied on the next keypress.

Reply:
Thanks @jonathonherbert, that makes sense. Is there any particular reason to keep them, would it not be sensible to have them removed when clicking into a blank editor?

Reply:
Ah, I didn't address your second point. Clicking into a blank field with 'bold' styling queued up (in storedMarks) and losing that styling as a result of the click would be incorrect behaviour. I only caught it 'cos an integration test failed 😅

Reply:
I would have thought it would desirable to remove the stored mark if I focus on an empty editor. I would think it's a new selection, separate to where you have queued up the styling?

Reply:
Discussed out of band – turns out stored marks were altered unnecessarily here, a good spot 🙏